compiler/optimizer/LoadExtensions.hpp

```diff
@@ -34,81 +34,210 @@
 #include "optimizer/Optimization_inlines.hpp" // for Optimization inlines
 #include "optimizer/OptimizationManager.hpp"  // for OptimizationManager
 
-class TR_BitVector;
-class TR_UseDefInfo;
-
+/** \brief
+ *     Examines how often a load is being used to feed into a signed vs. unsigned conversion (ex. i2l vs iu2l) and
+ *     attempts to skip the conversion by sign/zero extending the load at the point it is evaluated (the source).
```

Comment:
For extra clarity I'd add "(~the source~ where the soure value is actually loaded)"

Reply:
Fixed in 255151b